### ISSUE TITLE
Richtext editor scroll improvement

### ISSUE THREAD

**Issue by onno-timmerman** [wagtail/wagtail#7777]
When you edit longer text in the richtext editor and reach the bottom of your screen, the browser does not scroll along.  This can break the "zone" you are in when writing.

My suggestion is that the browser auto scrolls along while you type in the text editor.

**Comment by vihaanthora** [comment]
@onno-timmerman Hi, I am looking into this issue. The RichTextField inherits from models.TextField, which works perfectly fine in itself. It scrolls automatically and also breaks the line whenever the width is reached. The problem seems to be with the custom widget made for rich text editor, namely wagtail.admin.rich_text.get_rich_text_editor_widget.

**Comment by thibaudcolas** [comment]
As far as I can see the browser does scroll along regardless of which field type you use, it just does so with one more line’s worth of offset with `RichTextField` compared to `TextField`.

In both cases though, the area where you’re typing is covered by the fixed footer actions. And in both cases, it seems like this would be solved by adding a `scroll-padding-bottom` similarly to how we add top padding for the slim header: https://github.com/wagtail/wagtail/blob/main/client/scss/components/_grid.legacy.scss#L31

A `scroll-padding-bottom: 100px` seemed to do the trick for both TextField and RichTextField.

**Comment by AfiMaameDufie** [comment]
Hi @thibaudcolas 
Can I take a look at this?

**Comment by thibaudcolas** [comment]
@AfiMaameDufie Yes, no need to ask, just say you intend to work on this and give us a few notes of how you’ll approach the problem if poss!

**Comment by AfiMaameDufie** [comment]
Okay, thanks!

**Comment by AfiMaameDufie** [comment]
Hi @thibaudcolas, 
I went through the docs and created the wagtail site, added two blogs in attempt to replicate the problem mentioned above. However, I cannot see the issue with the scroll when editing. 
Can you give me the steps to produce the problem?

**Comment by 4the4ryushin** [comment]
> Hi @thibaudcolas Can I take a look at this?

@thibaudcolas are you still working on this?

**Comment by AfiMaameDufie** [comment]
@4the4ryushin you can work on it.

**Comment by 4the4ryushin** [comment]
Okay let me give it a try, thank you.

**Comment by salty-ivy** [comment]
> scroll-padding-bottom: 100px

this is definitely working but how would it even help it save draft box is already covering up the view 
![image](https://user-images.githubusercontent.com/74553951/200192246-a5c52daa-54bf-4ea2-b382-be34b9fe8078.png)

is there any other alternative ?or any way that this can be dealt with

**Comment by ArijeetBanerjee** [comment]
I want to work on this issue. kindly give me permission and access to work on this issue

**Comment by Aym-n** [comment]
@salty-ivy are you still working on this issue? and is this issue still open?

**Comment by ArijeetBanerjee** [comment]
Hey there 
It would be an honor to work in this issue
If it is still available, I'd be glad to work in it

**Comment by salty-ivy** [comment]
Yes I have given it a try tho, not complete there little occupied I would let you know if I won't be able to squeeze time for it, I would let you know.

Need few more days

**Comment by salty-ivy** [comment]
@Aym-n you can pick this issue if you want, I can't pursue it for now working on other issues.

**Comment by BykovVik** [comment]
I take on this task

**Comment by salty-ivy** [comment]
@BykovVik sure go ahead! Lemme know if you need any help

**Comment by BykovVik** [comment]
I'm sorry I can't do anything yet, I'll try again later

**Comment by sande265** [comment]
Hi if this issue is open & nobody's already working on it I would like to go on ahead with this issue.

**Comment by ssstain** [comment]
Right from the #support channel i was advised to open an issue if there is none yet. Here are some videos on the issue that might be helpful.

https://user-images.githubusercontent.com/15293404/220902217-61b5ba7b-6183-48fe-b884-de366f5aa562.mp4


https://user-images.githubusercontent.com/15293404/220902234-fab22cee-35af-4847-bab7-5850e7dd390a.mp4

**Comment by sande265** [comment]
Going ahead with the issue, heads up.

**Comment by sohankamble** [comment]
I want to work on this issues please assign me this issue

**Comment by gasman** [comment]
Thanks for your interest @sohankamble! We do not assign issues outside of the core team - anyone is welcome to work on any issue. As you can see above, @sande265 has already expressed an interest in working on this, so it's likely that you would be duplicating their existing effort. Nevertheless, if you're OK with this, you can proceed.

For further help in getting started, please see https://dev.to/lb/ten-tasty-ingredients-for-a-delicious-pull-request-hgc - you may also wish to join [our #new-contributors channel on Slack](https://github.com/wagtail/wagtail/wiki/Slack).

**Comment by abhi-kakadiya** [comment]
@sande265 are you still working on this issue? and is this issue still open?. I would like to go on ahead with this issue.

**Comment by sande265** [comment]
@abhi-kakadiya you can go ahead with the issue, check with @gasman if the issue has already been resolved else you can go ahead with the issue.

**Comment by gasman** [comment]
As far as I know this is still open, so go ahead!

**Comment by saurabhbakolia** [comment]
Hey @gasman  Can I work on it. As it is still open.

**Comment by gasman** [comment]
@saurabhbakolia As I said above, there's no need to seek permission - anyone is free to work on any issue.

**Comment by saurabhbakolia** [comment]
Thanks for mentioning.

**Comment by faishal882** [comment]
hey @saurabhbakolia  if the issue is solved please let me know or i can give it a try

**Comment by saurabhbakolia** [comment]
Hi, @faishal882 still trying it out. If failed to complete it. I will tell you.

**Comment by faishal882** [comment]
Okk thanks

**Comment by shyamraghu** [comment]
@gasman  Sir help me to regenerate this issue

**Comment by shamvrueth** [comment]
@thibaudcolas Can I work on this if its still not resolved?

**Comment by TejasSaraf** [comment]
I want to work on this issue please assign me this issue.
